drivers/clock_control/nrf_clock_calibration.c

```diff
@@ -190,16 +194,24 @@ static s16_t get_temperature(void)
  */
 static void measure_temperature(struct k_work *work)
 {
-	s16_t temperature;
+	s16_t temperature = 0;
 	s16_t diff;
 	bool started = false;
 	int key;
+	int rc;
 
-	temperature = get_temperature();
-	diff = abs(temperature - prev_temperature);
+	rc = get_temperature(&temperature);
 
 	key = irq_lock();
 
+	if (rc != 0) {
+		/* Temperature read failed: update state to try again */
+		cal_state = CAL_HFCLK_REQ;
```

Comment:
ok, i've tried it. What you can do here is to instead of setting `CAL_HFCLK_REQ` call `to_idle()`. That will restart calibration timer and make another attempt.

Reply:
Thanks; commit updated.